actioncable/lib/action_cable/engine.rb

```diff
@@ -34,5 +37,22 @@ class Railtie < Rails::Engine # :nodoc:
         options.each { |k,v| send("#{k}=", v) }
       end
     end
+
+    initializer "action_cable.routes" do
+      config.after_initialize do |app|
+        config = app.config
+        if config.action_cable.mounted
+          app.routes.prepend do
+            mount ActionCable.server => config.action_cable.url
+          end
+        end
+
+        ActionDispatch::Routing::RouteWrapper.class_eval do
+          class_attribute :action_cable_url
+          prepend ActionCable::RouteWrapper
+          self.action_cable_url = config.action_cable.url
```

Comment:
Would rather we capture the url and build the regex here. Maybe we can avoid adding a `class_attribute` altogether.